plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/edit.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { getSettingWithCoercion } from '@woocommerce/settings';
+import { isBoolean } from '@woocommerce/types';
+
+/**
+ * Internal dependencies
+ */
+import { useIsDescendentOfSingleProductBlock } from '../../atomic/blocks/product-elements/shared/use-is-descendent-of-single-product-block';
+import { AddToCartOptionsSettings } from './settings';
+import { INNER_BLOCKS_TEMPLATE } from './constants';
+export interface Attributes {
+	className?: string;
+	isDescendentOfSingleProductBlock: boolean;
+}
+
+export type FeaturesKeys = 'isBlockifiedAddToCart';
+
+export type FeaturesProps = {
+	[ key in FeaturesKeys ]?: boolean;
+};
+
+export type UpdateFeaturesType = ( key: FeaturesKeys, value: boolean ) => void;
+
+// Pick the value of the "blockify add to cart flag"
+const isBlockifiedAddToCart = getSettingWithCoercion(
+	'isBlockifiedAddToCart',
+	false,
+	isBoolean
+);
+
+const AddToCartOptionsEdit = ( props: BlockEditProps< Attributes > ) => {
+	const { setAttributes } = props;
+
+	const blockProps = useBlockProps( {
+		className: 'wc-block-add-to-cart-with-options-wrapper',
+	} );
+	const { isDescendentOfSingleProductBlock } =
+		useIsDescendentOfSingleProductBlock( {
+			blockClientId: blockProps?.id,
+		} );
+
+	useEffect( () => {
+		setAttributes( {
+			isDescendentOfSingleProductBlock,
+		} );
+	}, [ setAttributes, isDescendentOfSingleProductBlock ] );
```

Comment:
related to the block attributes comment above, I'd like to avoid this way to identify when the block instance is a descendant of the SingleProduct block.

Reply:
I created the issue https://github.com/woocommerce/woocommerce/issues/53284 to tackle this problem.